File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/domain/model/FizzBuzzValue.java
package fizzbuzz.domain.model;

import fizzbuzz.IFizzBuzz;

import java.util.ArrayList;
import java.util.List;

public class FizzBuzzValue implements IFizzBuzz {
  private final int _number;
  private final String _value;

  public FizzBuzzValue(int number, String value) {
    if (number < 0)
    throw new IllegalArgumentException(
      "FizzBuzzValue can't generate by minus number"
    );
    _number = number;
    _value = value;
  }

  public int getNumber() {
    return _number;
  }

  public String getValue() {
    return _value;
  }

  @Override
  public List<FizzBuzzValue> getList() {
    return new ArrayList<>();
  }

}

File: dev/20190822/k2works/src/FizzBuzzValue.ts
import { AssertionError } from "assert";
import { IFizzBuzz } from "./IFizzBuzz";

export class FizzBuzzValue implements IFizzBuzz {
  public get number(): number {
    return this._number;
  }
  public get value(): string {
    return this._value;
  }
  // tslint:disable-next-line: variable-name
  private _number: number;
  // tslint:disable-next-line: variable-name
  private _value: string;

  // tslint:disable-next-line: variable-name
  constructor(number: number, value: string) {
    if (number < 0) {
      throw new Error("FizzBuzzValue can't generate by minus number");
    }
    this._number = number;
    this._value = value;
  }
  public getValue(): string {
    return this.value;
  }
  public getList(): FizzBuzzValue[] {
    return [];
  }
}
